CHANGELOG.md

```diff
@@ -3,13 +3,14 @@ All notable changes to this project will be documented in this file.
 
 ## [Unreleased]
 
-- Add classes to cast ObjectId and UUID instances [#1](https://github.com/GromNaN/laravel-mongodb-private/pull/1) by [@alcaeus](https://github.com/alcaeus).
+- Add classes to cast `ObjectId` and `UUID` instances [#1](https://github.com/GromNaN/laravel-mongodb-private/pull/1) by [@alcaeus](https://github.com/alcaeus).
 - Add `Query\Builder::toMql()` to simplify comprehensive query tests [#6](https://github.com/GromNaN/laravel-mongodb-private/pull/6) by [@GromNaN](https://github.com/GromNaN).
 - Fix `Query\Builder::whereNot` to use MongoDB [`$not`](https://www.mongodb.com/docs/manual/reference/operator/query/not/) operator [#13](https://github.com/GromNaN/laravel-mongodb-private/pull/13) by [@GromNaN](https://github.com/GromNaN).
 - Fix `Query\Builder::whereBetween` to accept `Carbon\Period` object [#10](https://github.com/GromNaN/laravel-mongodb-private/pull/10) by [@GromNaN](https://github.com/GromNaN).
 - Throw an exception for unsupported `Query\Builder` methods [#9](https://github.com/GromNaN/laravel-mongodb-private/pull/9) by [@GromNaN](https://github.com/GromNaN).
 - Throw an exception when `Query\Builder::orderBy()` is used with invalid direction [#7](https://github.com/GromNaN/laravel-mongodb-private/pull/7) by [@GromNaN](https://github.com/GromNaN).
 - Throw an exception when `Query\Builder::push()` is used incorrectly [#5](https://github.com/GromNaN/laravel-mongodb-private/pull/5) by [@GromNaN](https://github.com/GromNaN).
+- Update `like` operators to be case-sensitive and improve support for `ilike`,  `regex`, `regexp` and their `not` variants [#17](https://github.com/GromNaN/laravel-mongodb-private/pull/17) by [@GromNaN](https://github.com/GromNaN).
```

Comment:
Something else we may want to consider is leaving "like" as-is for BC and _not_ implementing `ilike` (because it'd be redundant). We could then push `regex` as the explicit operator to use, and have it default to case-sensitive mode.

With that, there'd be no risk that an application's existing `like` query would start behaving differently. I realize the major version bump gives us the option to break BC, but this seems like something that will still go unnoticed by some fraction of users.

Do `regex` and `regexp` already exist in the library? If so, do they presently default ot case-insensitive matching?

I'd be in favor of removing the `regexp` alias if it only ever existed in the Laravel MongoDB library. If _that_ was removed, queries would stop working altogether, so I'm not worried about it going unnoticed.

Reply:
I reverted the `ilike` to stick with the case-insensitive `like`.